egov/egov-egi/src/main/java/org/egov/infra/admin/master/service/UserService.java

```diff
@@ -180,14 +190,72 @@ else if (user.getType().equals(UserType.EMPLOYEE))
         createUserRequest.setRequestInfo(microserviceUtils.createRequestInfo());
 
         final RestTemplate restTemplate = new RestTemplate();
+        UserDetailResponse udr = null;
         try {
-            restTemplate.postForObject(createUserServiceUrl, createUserRequest, UserDetailResponse.class);
+            udr = restTemplate.postForObject(createUserServiceUrl, createUserRequest, UserDetailResponse.class);
         } catch (final Exception e) {
             final String errMsg = "Exception while creating User in microservice ";
             LOGGER.error(errMsg, e);
             throw new ApplicationRuntimeException(errMsg, e);
         }
-        return userRepository.save(user);
+        final Long userId = udr.getUser().get(0).getId();
```

Comment:
We need to retain the same user id which was generated while creating user in M@S user service. To avoid PK generation in ERP while creating user, the JDBC query is used. This is used  only when M@S and ERP coexistence exists. Otherwise  userRepository.save(user);  will be used.  Is it okay if we move this code which does JDBC call to create user to a MicroserviceUtils and call it from here?